src/kr_comp.cpp

```diff
@@ -0,0 +1,104 @@
+#include <Rcpp.h>
+#define INCLUDE_RCPP
+#include "utils.h"
+#include "covariance.h"
+#include "chol_derivative.h"
+
+using namespace Rcpp;
+using std::string;
+
+vector<double> as_vector(NumericVector input) {
```

Comment:
this whole business of converting Rcpp types to std types would fit better in utils file I think

Reply:
utils.h is a header file and I am currently putting the declaration in chol_derivative.h file. The definition need to be in cpp files (or we need another file called utility.cpp? this is different from utils.h).

Reply:
Yeah we can have new utility.cpp

Reply:
updated